packages/web3-hooks/base/src/useFungibleToken.ts

```diff
@@ -16,6 +16,15 @@ export function useFungibleToken<S extends 'all' | void = void, T extends Networ
 
     return useAsyncRetry<Web3Helper.FungibleTokenScope<S, T> | undefined>(async () => {
         if (!hub) return
-        return attemptUntil([() => hub?.getFungibleToken?.(address ?? '', { chainId })], fallbackToken)
+        return attemptUntil(
+            [
+                async () => {
+                    const token = await hub?.getFungibleToken?.(address ?? '', { chainId })
```

Comment:
```ts
if (!hub || !address) return
```

Reply:
`options` should be removed from the dependency array, since this hook should not depend on `options`.

`hub` is already in the dependency array. If `options`'s value changes, the `hub` will changes, and this hook will re-execute.

Reply:
Here is my suggestion patch

```diff
diff --git a/packages/web3-hooks/base/src/useFungibleToken.ts b/packages/web3-hooks/base/src/useFungibleToken.ts
index d0a50f7db..b62f3449f 100644
--- a/packages/web3-hooks/base/src/useFungibleToken.ts
+++ b/packages/web3-hooks/base/src/useFungibleToken.ts
@@ -15,7 +15,7 @@ export function useFungibleToken<S extends 'all' | void = void, T extends Networ
     const chainId = useChainId(pluginID, options?.chainId)

     return useAsyncRetry<Web3Helper.FungibleTokenScope<S, T> | undefined>(async () => {
-        if (!hub) return
-        return attemptUntil([() => hub?.getFungibleToken?.(address ?? '', { chainId })], fallbackToken)
-    }, [address, hub, chainId, JSON.stringify(options)])
+        if (!hub || !address) return
+        return attemptUntil([() => hub?.getFungibleToken?.(address, { chainId })], fallbackToken)
+    }, [address, hub, chainId])
 } 
```

Reply:
done